sympy/integrals/integrals.py

```diff
@@ -444,6 +444,12 @@ def doit(self, **hints):
             function = function.doit(**hints)
         if function.is_zero:
             return S.Zero
+        if len(self.limits) == 1 and len(self.limits[0]) == 3:
+            (x, a, b) = self.limits[0]
+            if any(i.is_infinite for i in (a,b)) and \
```

Comment:
We should check the signs of the limits. If the lower limit is `oo` or the upper limit is `-oo` then the integral will need to be flipped and the result will be negated.

Reply:
Yeah i just figured that out, i have made changes for that.